aws/ec2.go

```diff
@@ -0,0 +1,225 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultTimeOut = 20
+	defaultRetries = 3
+)
+
+func convertError(err error) string {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return fmt.Sprintf("%s: %s", awsErr.Code(), awsErr.Message())
+	}
+
+	return fmt.Sprintf("%v", err.Error())
+}
+
+func retriable(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		switch awsErr.Code() {
+		case "InternalError":
+			return true
+		case "InternalFailure":
+			return true
+		case "RequestLimitExceeded":
+			return true
+		case "ServiceUnavailable":
+			return true
+		case "Unavailable":
+			return true
+		}
+	}
+
+	return false
+}
+
+func checkSourceDestinationValueIsDisable(check string) bool {
+	return check == "disable"
+}
+
+func UpdateSrcDstCheck(check string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeOut*time.Second)
+	defer cancel()
+
+	awsSession, err := awssession.NewSession()
+	if err != nil {
+		return fmt.Errorf("error creating AWS session: %v", err)
+	}
+
+    cli, err := NewEc2MetadataClient(awsSession)
+	if err != nil {
+		return err
+	}
+
+	if !cli.ec2MetadataSvc.AvailableWithContext(ctx) {
+		return fmt.Errorf("EC2 metadata service is unavailable or not running on an EC2 instance")
+	}
+
+	ec2Region, err := cli.getEC2Region(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting ec2 region: %s", convertError(err))
+	}
+
+	ec2Id, err := cli.getEC2InstanceID(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting ec2 instance-id: %s", convertError(err))
+	}
+
+	ec2Cli, err := NewEc2Client(awsSession, ec2Region, ec2Id)
+	if err != nil {
+		return err
+	}
+
+	ec2NetId, err := ec2Cli.getEC2NetworkInterfaceId(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting ec2 network-interface-id: %s", convertError(err))
+	}
+
+	checkVal := !checkSourceDestinationValueIsDisable(check)
+	err = ec2Cli.setEC2SourceDestinationCheck(ctx, ec2NetId, checkVal)
+	if err != nil {
+		return fmt.Errorf("error setting src-dst-check for network-interface-id: %s", convertError(err))
+	}
+
+	log.Infof("Successfully set source-destination-check to %t on network-interface-id: %s", checkVal, ec2NetId)
+	return nil
+}
+
+type ec2MetadataClient struct {
+	ec2MetadataSvc *ec2metadata.EC2Metadata
+}
+
+func NewEc2MetadataClient(awsSession *awssession.Session) (*ec2MetadataClient, error) {
```

Comment:
Also, I think this is just an internally used function, so we should probably make it private.